common4j/src/main/com/microsoft/identity/common/java/commands/parameters/nativeauth/SignUpStartUsingPasswordCommandParameters.java

```diff
@@ -0,0 +1,47 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.commands.parameters.nativeauth;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * A set of Sign Up Start command parameters for sending the start request to trigger the sign up flow but using email password.
+ * extends from {@link BaseSignUpStartCommandParameters}
+ */
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@SuppressFBWarnings("EI_EXPOSE_REP2")   //Suppresses spotbugs warning on the builder class
+@SuperBuilder(toBuilder = true)
+public class SignUpStartUsingPasswordCommandParameters extends BaseSignUpStartCommandParameters {
+    private static final String TAG = SignUpStartUsingPasswordCommandParameters.class.getSimpleName();
+
+    /**
+     * The password of the user.
+     */
+    @NonNull
+    public final char[] password;
```

Comment:
Do we also need to declare e-mail here? Or is that part of the parent class already?

Reply:
the username (email address) is part of base class. `BaseSignUpStartCommandParameters`